source/interact-data/transaction.txt

```diff
@@ -0,0 +1,283 @@
+
+- :ref:`mongoid-txn-high-level`: {+odm+} manages the life cycle of the
+  transaction. You can use this API in {+odm+} v9.0 and later.
+
+- :ref:`mongoid-txn-low-level`: You must manage the life cycle of the
+  transaction. You can use this API in {+odm+} v6.4 and later.
+
+The :ref:`mongoid-txn-session` section describes how to make changes to
+your data from within a session without performing a transaction.
+
+.. _mongoid-txn-high-level:
+
+Higher Level API
+----------------
+
+You can start a transaction by calling the ``transaction()`` method on
+an instance of a model, on the model class, or on a ``{+odm+}`` module.
+
+When you call the ``transaction()`` method, {+odm+} performs the
+following tasks:
+
+1. Creates a session on the client.
+
+#. Starts a transaction on the session.
+
+#. Performs the specified data changes.
+
+#. Commits the transaction if no errors occur or ends the transaction if
+   there is an error. See the :ref:`` section to learn more an
+
+#. Closes the session.
+
+If your transaction is committed, {+odm+} calls any ``after_commit``
+callbacks for all objects modified inside the transaction. If there is
+an error and the transaction is rolled back, {+odm+} calls any
+``after_rollback`` callbacks for all objects modified inside the
+transaction.
+
+Example
+~~~~~~~
+
+This example uses the following models to represent documents that
+describe books and films:
+
+.. literalinclude:: /includes/interact-data/transaction.rb
+   :start-after: start-example-models
+   :end-before: end-example-models
+   :language: ruby
+   :dedent:
+
+The following code demonstrates how to perform a transaction on
+different objects to change data in multiple collections:
+
+.. literalinclude:: /includes/interact-data/transaction.rb
+   :start-after: start-txn-operations
+   :end-before: end-txn-operations
+   :language: ruby
+   :dedent:
+
+Client Limitations
+~~~~~~~~~~~~~~~~~~
+
+Only operations on the same client are in scope of a transaction,
+because each transaction is attached to a specific client. Ensure
+that you use objects from same client inside the transaction method
+block.
+
+The following example defines model classes that use different clients
+and demonstrates how operations are run based on the origin client:
+
+.. literalinclude:: /includes/interact-data/transaction.rb
+   :start-after: start-different-clients
+   :end-before: end-different-clients
+   :language: ruby
+   :dedent:
+
+.. note::
+   
+   When you call the ``transaction()`` method on the ``{+odm+}`` module,
+   {+odm+} creates the transaction by using the ``:default`` client.
+
+Ending Transactions
+~~~~~~~~~~~~~~~~~~~
+
+Any exception raised inside the transaction method block ends the
+transaction and rolls back changes. Usually, {+odm+} displays the
+exception, except for the ``Mongoid::Errors::Rollback`` exception. Implement
+this exception in your application to explicitly end the transaction
+without passing on an exception.
```

Comment:
I'm a bit confused on this exception. A couple questions that come up for me:
- What is the difference between it and other exceptions that also end transactions and roll back changes? 
- Does it only happen when explicitly implemented? 
- What does "without passing on an exception" mean?

Reply:
It seems like raising this error would be useful for a situation where you check for a certain condition and if its not met, you want the transaction to be aborted and changes to be rolled back. You want this to happen "silently" ie without erroring and seeing a specific error. I am reading the API docs and will try to reword.